File: LeetCode/168. Excel Sheet Column Title/Solution.java
public class Solution {
    public String convertToTitle(int n) {

        final char[] DIGITS = {
                'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
                'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'
        };

        int l = DIGITS.length;
        String result = "";

        while (n > 0) {
            int i = n % l;

            switch (i) {
                case 0:
                    result = DIGITS[l - 1] + result;
                    break;
                default:
                    result = DIGITS[i - 1] + result;
            }

            n = (n - 1) / l;
        }

        return result;
    }
}

File: LeetCode/168. Excel Sheet Column Title/solution.js
/**
 * @param {number} n
 * @return {string}
 */

const digits = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
const length = digits.length;

const convertToTitle = function(n) {
  let result = '';

  while (n > 0) {
    const p = n % length;
    result = digits.charAt(!!p ? p - 1 : length - 1) + result;
    n = Math.floor((n - 1) / length);
  }

  return result;
};

// charCode: 65 - 90
const BASE = 26;
const CODE = 64;
const convertToTitle = function(n) {
  let result = '';
  while (n > 0) {
    const remainder = n % BASE;
    const i = !remainder ? BASE : remainder;
    result = String.fromCharCode(i + CODE) + result;
    n = (n - i) / BASE;
  }
  return result;
};
